testing/trino-testing/src/main/java/io/trino/testing/AbstractTestFailureRecovery.java

```diff
@@ -206,7 +206,7 @@ protected void testSelect(String query, Optional<Session> session, Consumer<Quer
                 .experiencing(TASK_GET_RESULTS_REQUEST_TIMEOUT)
                 // using boundary stage so we observe task failures
                 .at(boundaryDistributedStage())
-                .failsWithoutRetries(failure -> failure.hasMessageContaining("Encountered too many errors talking to a worker node"))
+                .failsWithoutRetries(failure -> failure.hasMessageFindingMatch("Encountered too many errors talking to a worker node|Error closing remote buffer.*3 failures"))
```

Comment:
add comment why we expect both?

Reply:
It depends on which communication path gets "killed" by injected failure first. I cannot think of any really helpful comment here. But maybe @arhimondr have an idea.